include/rnp/rnp.h

```diff
@@ -417,6 +430,93 @@ RNP_API rnp_result_t rnp_supports_feature(const char *type, const char *name, bo
  */
 RNP_API rnp_result_t rnp_supported_features(const char *type, char **result);
 
+/**
+ * @brief Add new security rule to the FFI. Security rules allows to override default algorithm
+ *        security settings by disabling them or marking as insecure. After creation of FFI
+ *        object default rules are added, however caller may add more strict rules or
+ *        completely overwrite rule table by calling rnp_remove_security_rule().
+ *
+ * @param ffi initialized FFI object.
+ * @param type type of the feature, cannot be NULL. Currently only RNP_FEATURE_HASH_ALG is
+ *             supported.
+ * @param name name of the feature, i.e. SHA1, MD5. The same values are used in
+ *             rnp_supports_feature()/rnp_supported_features().
+ * @param flags additional flags. Following ones currently supported:
+ *              - RNP_SECURITY_OVERRIDE : override all other rules for the specified feature.
+ *                May be used to temporary enable or disable some feature value (like, allow to
+ *                verify SHA1 or MD5 signature), and then revert changes via
+ *                rnp_remove_security_rule().
+ * @param from timestamp, since which rule is active. Objects which has creation time (like
+ *             signatures) are matched with the closest rules from the past, unless there is
+ *             rule with override flag.
+ * @param level security level of the rule. Currently following ones are defined:
+ *              - RNP_SECURITY_DISABLED : feature (for instance, MD5 algorithm) is completely
```

Comment:
RNP_SECURITY_DISABLED macro somehow implies security is disabled, so MD5 algo should pass, but description says otherwise :) Maybe some better macro name can be used here, like RNP_SECURITY_PROHIBITED

Reply:
Thanks, naming this looks better indeed.